### PR TITLE
Advanced MMG

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Adds the advanced MMG, an automatic version of the deployable turret.

This one uses signals and stuff, so it's more responsive.

dreamseeker_MOIfJDlMUS

## Changelog

:cl:
add: T-90 Mounted Machine Gun
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammunition/_firing.dm [hunk 1]
```diff
@@ -26,7 +26,7 @@
 		user.newtonian_move(get_dir(target, user))
 	else if(ismovable(fired_from))
 		var/atom/movable/firer = fired_from
-		if(!firer.newtonian_move(get_dir(target, fired_from), instant = TRUE))
+		if(!firer.anchored && !firer.newtonian_move(get_dir(target, fired_from), instant = TRUE)) // SKYRAT EDIT CHANGE
```
Comment (Zonespace27) [inline code/modules/projectiles/ammunition/_firing.dm]: smh, i still say this is a feature

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+/obj/machinery/mounted_machine_gun/update_overlays()
+	. = ..()
+	if(ammo_box)
+		. += "ammo_box"
+	if(cover_open)
+		. += "cover_open"
+
+	switch(barrel_heat)
+		if(50 to 75)
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]: make these a define

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+/obj/machinery/mounted_machine_gun/Destroy()
+	QDEL_NULL(ammo_box)
+	QDEL_NULL(particles)
+	if(current_user)
+		unregister_mob(current_user)
+		current_user = null
+	return ..()
+
+/obj/machinery/mounted_machine_gun/process(delta_time)
+	if(barrel_heat > 0)
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]:
```suggestion
	if(barrel_heat)
```

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+		return
+	var/obj/undeployed_object = new spawned_on_undeploy(src)
+	//Keeps the health the same even if you redeploy the gun
+	undeployed_object.modify_max_integrity(max_integrity)
+	qdel(src)
+
+//BUCKLE HOOKS
+/obj/machinery/mounted_machine_gun/unbuckle_mob(mob/living/buckled_mob, force = FALSE, can_fall = TRUE)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, TRUE)
+	for(var/obj/item/I in buckled_mob.held_items)
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]: one letter var

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+		if(istype(I, /obj/item/gun_control))
+			qdel(I)
+	if(istype(buckled_mob))
+		buckled_mob.pixel_x = buckled_mob.base_pixel_x
+		buckled_mob.pixel_y = buckled_mob.base_pixel_y
+		if(buckled_mob.client)
+			buckled_mob.client.view_size.resetToDefault()
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]:
```suggestion
			buckled_mob?.client.view_size.resetToDefault()
```
remove above line

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+		return
+
+	COOLDOWN_START(src, trigger_cooldown, fire_delay)
+
+	playsound(src, firesound, 100)
+	casing.forceMove(drop_location()) //Eject casing onto ground.
+	casing.bounce_away(TRUE)
+
+	barrel_heat += barrel_heat_per_shot
+	if(barrel_heat >= 100)
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm]: define

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+	if(user.incapacitated())
+		return
+	setDir(get_dir(src,targeted))
+	user.setDir(dir)
+	switch(dir)
+		if(NORTH)
+			layer = BELOW_MOB_LAYER
+			plane = GAME_PLANE
+			user.pixel_x = 0
+			user.pixel_y = -14
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm:393]: make these defines maybe

File: modular_skyrat/modules/mounted_machine_gun/code/ammobox.dm [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+/obj/item/ammo_casing/b50cal
+	name = ".50 BMG bullet casing"
+	icon = 'modular_skyrat/modules/gunsgalore/icons/ammo/ammo.dmi'
+	icon_state = "792x57-casing"
+	caliber = CALIBER_50BMG
+	projectile_type = /obj/projectile/bullet/c50cal
+
+/obj/projectile/bullet/c50cal
+	name = ".50 BMG bullet"
+	damage = 35
```
Comment (Zonespace27) [inline modular_skyrat/modules/mounted_machine_gun/code/ammobox.dm:20]: i get the balance concerns - just find it a bit funny how a 10mm magnum round does 5 more damage than a 50 caliber machine gun

Comment (Stalkeros2) [reply modular_skyrat/modules/mounted_machine_gun/code/ammobox.dm:20]: This HMG is (was) kinda specially tailored for the Black Mesa away mission, so we had to sacrifice realism for balance and fun gameplay.

File: modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm (hunk 7 above)
Comment (Gandalf2k15) [reply modular_skyrat/modules/mounted_machine_gun/code/mounted_machine_gun.dm:393]: eh, too extra

**Comment by Gandalf2k15** [comment]
Merging, will have experimental version up shortly.

**Pull request merged**